client/my-sites/plugins/plugin-details-CTA/CTA-button.jsx

```diff
@@ -104,13 +114,27 @@ export default function CTAButton( { plugin, hasEligibilityMessages, disabled }
 	const pluginsPlansPageFlag = isEnabled( 'plugins-plans-page' );
 	const pluginsPlansPage = `/plugins/plans/${ plugin.slug }/yearly/${ selectedSite?.slug }`;
 
+	const getButtonText = () => {
+		if ( buttonText ) return buttonText;
+
+		if ( isMarketplaceProduct || isPreinstalledPremiumPlugin ) {
+			return translate( 'Purchase and activate' );
+		}
+
+		if ( shouldUpgrade ) {
+			return translate( 'Upgrade and activate' );
+		}
+
+		return translate( 'Install and activate' );
+	};
+
 	return (
 		<>
 			<PluginCustomDomainDialog
 				onProceed={ () => {
 					if ( hasEligibilityMessages ) {
-						if ( pluginsPlansPageFlag && shouldUpgrade ) {
-							return page( pluginsPlansPage );
+						if ( plansPage && pluginsPlansPageFlag && shouldUpgrade ) {
+							return page( `${ pluginsPlansPage }?pluginSlug=${ plugin.slug }` );
```

Comment:
https://github.com/Automattic/wp-calypso/pull/68154/files?w=1#diff-d5ac7cfa2d86b94db92445b2f1e3f311cafbe173848e5b2cc4dd86d550f90ae6R115

```suggestion
							return page( `${ pluginsPlansPage }` );
```

Reply:
Sounds good, [fixed here](https://github.com/Automattic/wp-calypso/pull/68154/commits/9a4db17bd1d9b8c6c354e04c3e4b6722b0c0c6e4)